iter/channel_test.go

```diff
@@ -40,6 +41,24 @@ func TestFromChannel(t *testing.T) {
 	assert.True(t, numbers.Next().IsNone())
 }
 
+func TestFromChannelFilter(t *testing.T) {
+	ch := make(chan int)
+
+	go func() {
+		defer close(ch)
+		ch <- 1
+		ch <- 2
+		ch <- 3
+	}()
+
+	numbers := iter.FromChannel(ch)
+	defer numbers.Collect()
```

Comment:
The reason this collect is here in other tests is because the underlying channel isn't closed until all items are fetched from the channel. In this case you're fetching all the items anyway so no need to `Collect` again. You can delete this line.